### ISSUE TITLE
speed up scipy optimize

### ISSUE THREAD

**Issue by jacobpennington** [jacobpennington/nems-lite#25]
Try out the suggestion here re: using JAX library to compute cost function gradient and providing that information to scipy.
(for their specific example, quoting a ~5000x speedup)
[https://stackoverflow.com/questions/68507176/faster-scipy-optimizations](https://stackoverflow.com/questions/68507176/faster-scipy-optimizations)

**Comment by jacobpennington** [comment]
Follow-up note: looks like there's even a `jax.scipy.optimize.minimize` function, so I'm hopeful this would slot in without too many headaches.

**Comment by jacobpennington** [comment]
Notes on Numba: doesn't seem like it will be worth it for most of the core Layers. Numpy functions that are already heavily optimized (like `np.dot` and `np.convolve`) already perform about as well or better. We might see a big benefit for slow Layers with lots of for loops and plain python (like STP), but still need to test that.
